lib/main.go

```diff
@@ -214,7 +215,29 @@ func (obj *Main) Run() error {
 		return fmt.Errorf("hostname cannot be empty")
 	}
 
+	// Get current user running mgmt
+	user, err := user.Current()
+	if err != nil {
+		return errwrap.Wrapf(err, "can't get current user")
+	}
+
+	// Use systemd SateDirectory if set. If not, use
+	// XDG_CACHE_DIR unless user is root, then use
+	// /var/lib/mgmt
 	var prefix = fmt.Sprintf("/var/lib/%s/", obj.Program) // default prefix
+	stateDir := os.Getenv("STATE_DIRECTORY")
+	xdg := os.Getenv("XDG_CACHE_HOME")
+
+	if xdg == "" {
+		xdg = fmt.Sprintf("%s/.cache/%s", user.HomeDir, obj.Program)
```

Comment:
What if HomeDir is "" ?

Reply:
What do you think is an appropriate dir for this? Something in /tmp?

Reply:
`if xdg == "" && HomeDir != "" {` ?

Reply:
Oh I see what you mean. I thought you meant assign one if they didn't have one.